ui/pages/institutional/custody/custody.js

```diff
@@ -0,0 +1,677 @@
+import React, {
+  useState,
+  useMemo,
+  useCallback,
+  useEffect,
+  useContext,
+} from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { debounce } from 'lodash';
+import { v4 as uuidv4 } from 'uuid';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  ButtonIcon,
+  Button,
+  Text,
+  Label,
+  IconName,
+  IconSize,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+} from '../../../components/component-library';
+import {
+  AlignItems,
+  DISPLAY,
+  FLEX_DIRECTION,
+  FONT_WEIGHT,
+  Color,
+  JustifyContent,
+  BorderRadius,
+  BorderColor,
+  BLOCK_SIZES,
+  TextColor,
+  TEXT_ALIGN,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import Box from '../../../components/ui/box';
+import {
+  CUSTODY_ACCOUNT_DONE_ROUTE,
+  DEFAULT_ROUTE,
+} from '../../../helpers/constants/routes';
+import {
+  getCurrentChainId,
+  getProvider,
+  getMmiConfiguration,
+} from '../../../selectors';
+import CustodyAccountList from '../connect-custody/account-list';
+import JwtUrlForm from '../../../components/institutional/jwt-url-form';
+
+const CustodyPage = () => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const trackEvent = useContext(MetaMetricsContext);
+  const dispatch = useDispatch();
+
+  const mmiActions = mmiActionsFactory();
+  const currentChainId = useSelector(getCurrentChainId);
+  const provider = useSelector(getProvider);
+  const { custodians } = useSelector(getMmiConfiguration);
+
+  const [selectedAccounts, setSelectedAccounts] = useState({});
+  const [selectedCustodianName, setSelectedCustodianName] = useState('');
+  const [selectedCustodianImage, setSelectedCustodianImage] = useState(null);
+  const [selectedCustodianDisplayName, setSelectedCustodianDisplayName] =
+    useState('');
+  const [selectedCustodianType, setSelectedCustodianType] = useState('');
+  const [connectError, setConnectError] = useState('');
+  const [currentJwt, setCurrentJwt] = useState('');
+  const [selectError, setSelectError] = useState('');
+  const [jwtList, setJwtList] = useState([]);
+  const [apiUrl, setApiUrl] = useState('');
+  const [addNewTokenClicked, setAddNewTokenClicked] = useState(false);
+  const [chainId, setChainId] = useState(0);
+  const [connectRequest, setConnectRequest] = useState(undefined);
+  const [accounts, setAccounts] = useState();
+
+  const handleConnectError = useCallback(
```

Comment:
I'm not asking for a change here i'm just curious what led to the usage of useCallback and useMemo in this component

Reply:
Lint was giving warnings, telling to add it :)